v3/lints/cabf_smime_br/lint_single_email_if_present.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * ZLint Copyright 2023 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package cabf_smime_br
+
+import (
+	"fmt"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+func init() {
+	lint.RegisterLint(&lint.Lint{
+		Name:          "e_single_email_if_present",
+		Description:   "If present, the subject:emailAddress SHALL contain a single Mailbox Address",
+		Citation:      "7.1.4.2.h",
+		Source:        lint.CABFSMIMEBaselineRequirements,
+		EffectiveDate: util.CABF_SMIME_BRs_1_0_0_Date,
+		Lint:          func() lint.LintInterface { return &singleEmailIfPresent{} },
+	})
+}
+
+type singleEmailIfPresent struct{}
+
+func NewSingleEmailIfPresent() lint.LintInterface {
+	return &singleEmailIfPresent{}
+}
+
+func (l *singleEmailIfPresent) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && c.EmailAddresses != nil && len(c.EmailAddresses) != 0
```

Comment:
Just want to check that this last part `len(c.EmailAddresses) != 0` is correct. It would seem that, if present, the field shouldn't be empty either. (I suspect though that this could just be an idiosyncrasy with Go X.509 and how fields are parsed)

Reply:
> I suspect though that this could just be an idiosyncrasy with Go X.509 and how fields are parsed

I precisely wanted to avoid any such shenanigans being possible. I was uncertain what upstream systems would generate in a `not present` scenario. I reckoned that both `nil` and empty would count as `not being present` (generally speaking, rather than speaking to the specific Go implementation).